### ISSUE TITLE
Schema is not reloaded on OAuth2 Authorization

### ISSUE THREAD

**Issue by BramEsposito** [tfranzel/drf-spectacular#1190]
**Describe the bug**
I am running:

- Django 4.2.10
- django-oauth-toolkit 2.3.0
- djangorestframework 3.14.0
- drf-spectacular 0.27.1

I have configured drf-spectacular to authorize with OAuth2, which appears as an option in the Swagger-ui authorize popup.
However, after authorizing, the schema is not reloaded.

**To Reproduce**
my spectacular settings are as follows: 
```(python)
SPECTACULAR_SETTINGS = {
   "TITLE": "Your Project API",
    "DESCRIPTION": "Your project description",
    "VERSION": "1.0.0",
    "SERVE_INCLUDE_SCHEMA": False,
    "SERVE_PUBLIC": False,
    # OTHER SETTINGS
    "OAUTH2_FLOWS": ["clientCredentials"],
    "OAUTH2_AUTHORIZATION_URL": "/o/authorize/",
    "OAUTH2_TOKEN_URL": "/o/token/",
}
```

**Expected behavior**
I expect the schema to reload after authenticating successfully with OAuth2 as per the ```reloadSchemaOnAuthChange``` plugin

**Comment by tfranzel** [comment]
yes it should reload. not exactly sure why. I have only tested this for non-oauth cases myself.

* is there something in the browser console? warnings?
* has this ever worked before for you? 
* are you sure it is not reloading? maybe it is reloading but the credentials are not injected?

**Comment by BramEsposito** [comment]
Wow, you respond faster than I anticipated.

I am authorize and reload with other authorization methods, but not with OAuth

I have solved this for myself by customising the swagger_ui plugin and adding an additional actionwrapper:

`
const reloadSchemaOnAuthChange = () => {
  return {
    statePlugins: {
      auth: {
        wrapActions: {
// ADD THIS BLOCK
          authorizeOauth2:(ori) => (...args) => {
            schemaAuthFailed = false;
            setTimeout(() => ui.specActions.download());
            return ori(...args);
          },
// END OF ADDED BLOCK
          authorize: (ori) => (...args) => {
            schemaAuthFailed = false;
            setTimeout(() => ui.specActions.download());
            return ori(...args);
          },
          logout: (ori) => (...args) => {
            schemaAuthFailed = false;
            setTimeout(() => ui.specActions.download());
            return ori(...args);
          },
        },
      },
    },
  };
};
`

**Comment by tfranzel** [comment]
Excellent! Of course they used another method. doh.

Do you want to make a PR or should I add it when I come around to it?
